sdk/storage/Azure.Storage.Blobs/src/BlobBaseClient.cs

```diff
@@ -686,6 +709,15 @@ private void SetNameFieldsIfNull()
                         Pipeline.ResponseClassifier,
                         Constants.MaxReliabilityRetries);
 
+                    // if using clientside encryption, wrap the auto-retry stream in a decryptor
+                    // we already return a nonseekable stream; returning a crypto stream is fine
+                    if (UsingClientSideEncryption)
+                    {
+                        stream = await ClientSideDecryptInternal(stream, response.Value.Metadata, encryptedRange.OriginalRange, response.Value.ContentRange, async, cancellationToken).ConfigureAwait(false);
```

Comment:
maybe we could make this extension and move out crypto code from this client ?

Reply:
Created new classes to handle the translation between clients and service-agnostic encryption/decryption logic. Upcoming push.